Source/Logger/Generic/generic_logger.cpp

```diff
@@ -0,0 +1,26 @@
+#include "pch.h"
+
+#include "../trace_internal.h"
+
+uint64_t Internal_ThisThreadId() noexcept
+{
+    auto tidi = GetThreadIdInfo();
+    if (tidi.callback)
```

Comment:
>tidi.callback [](start = 8, length = 13)

can these callbacks be set at runtime? If so you should have to do something like
auto callback = foo.callback
if (callback)
{
    callback(...)
}

Reply:
here and below

---
In reply to: [239286364](https://github.com/Microsoft/libHttpClient/pull/340#discussion_r239286364) [](ancestors = 239286364)

Reply:
auto tidi is making a copy of the object returned by GetThreadInfo();
(auto& would preserve the reference)